### PR TITLE
Bump to 3.1.1, add missing dependencies, metadata and maintainer

### PR DESCRIPTION
**Pull request by bollwyvl**
For #28. 

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- adds `keyring` and `importlab_metadata` (forcing the removal of `noarch: python`)
- updates the license string to be SPDX
- adds docs url and other metdata
- adds myself (@bollwyvl) as a maintainer

Given the upstream has been pretty free with adding deps, I'd like to see this add the tests as well, as they are neatly packaged in the distribution: there is at least one missing test dependency (`jaraco.envs`) so I haven't add it here, but will take a look at getting those built.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,16 +37,20 @@ test:
...
+  license: Apache-2.0
   license_family: Apache
   license_file: LICENSE
   summary: 'Collection of utilities for interacting with PyPI'
+  doc_url: https://twine.readthedocs.io
+  doc_source_url: https://github.com/pypa/twine/tree/master/docs
 
 extra:
   recipe-maintainers:
     - jankatins
```
Comment (jankatins) [inline recipe/meta.yaml]: I think it is time to say goodbye here: can you remove myself from the maintainer list?

Comment (bollwyvl) [reply recipe/meta.yaml]: :crying_cat_face: thanks for all the packaging!

Comment (bollwyvl) [reply recipe/meta.yaml]: removed. maybe drop a note down in the main body of the issue as well (code comments get hard to find sometimes)

**Comment by bollwyvl** [comment]
Thanks!

**Pull request merged**